### DISCUSSION TITLE
Customise displayed value whilst maintaining original data type for sorting, filtering, etc. (_psp_repr_ & _psp_dtype_)

### DISCUSSION THREAD

**Discussion by jonw09** [perspective-dev/perspective#2233] [Q&A]
## Support Question

I'm looking for a way to customise the value shown in the datagrid but maintain the underlying type in the Table so that filtering, sorting, etc. can be applied correctly to the column.

For example, if I had a column which contained prices I may wish to show a value such as `$1,000` however the underlying value should remain as `1000` to retain the ability to sort, filter, etc.

I've tried using various combinations of `_psp_repr_` and `_psp_dtype_`, however I can't seem to find a way to preserve the underlying data format at the same time as modifying the visible value on-screen. e.g.

```
class Price:
    def __init__(self, px):
        self._value = px

    def _psp_repr_(self):
        return "$%.2f" % self._value

    def _psp_dtype_(self):
        return float

    def __float__(self):
        return self._value
```

This correctly renders the string values in the grid, however if I try to apply a `>` filter to the column it's unsupported since the datatype is now a `str`.

Is this a supported setup or going to be supported in future?

**Comment by texodus** [comment]
@YUNGU23 Don't post chatbot content.

**Comment by texodus** [comment]
@jonw09 You can achieve this and much more with an expression column - no python needed:

```concat('$', string("Price"))```
